tests/integration/golang/admin/namespace/update_test.go

```diff
@@ -0,0 +1,150 @@
+		Code:                "default",
+		DefaultExperimentID: common.GetPointer(int32(0)),
+	})
+	assert.Nil(s.T(), err)
+	ns, err := s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+		ID:                  2,
+		Code:                "test2",
+		Description:         "test namespace 2 description",
+		DefaultExperimentID: common.GetPointer(int32(0)),
+	})
+	assert.Nil(s.T(), err)
+
+	request := request.Namespace{
+		Code:        "test2Updated",
+		Description: "test namespace 2 description updated",
+	}
+	assert.Nil(
+		s.T(),
+		s.AdminClient.WithMethod(
+			http.MethodPut,
+		).WithRequest(
+			request,
+		).DoRequest("/namespaces/%d", ns.ID),
+	)
+
+	namespaces, err := s.NamespaceFixtures.GetNamespaces(context.Background())
+	assert.Nil(s.T(), err)
+
+	// Check that the namespace has been updated
+	for _, namespace := range namespaces {
+		if namespace.ID == ns.ID {
+			assert.Equal(s.T(), namespace.Code, request.Code)
+			assert.Equal(s.T(), namespace.Description, request.Description)
+		}
+	}
+}
+
+func (s *UpdateNamespaceTestSuite) Test_Error() {
+	defer func() {
+		assert.Nil(s.T(), s.NamespaceFixtures.UnloadFixtures())
+	}()
+	_, err := s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+		ID:                  1,
+		Code:                "default",
+		DefaultExperimentID: common.GetPointer(int32(0)),
+	})
+	assert.Nil(s.T(), err)
+	_, err = s.NamespaceFixtures.CreateNamespace(context.Background(), &models.Namespace{
+		ID:                  2,
+		Code:                "test2",
+		Description:         "test namespace 2 description",
+		DefaultExperimentID: common.GetPointer(int32(0)),
+	})
+	assert.Nil(s.T(), err)
+	namespaces, err := s.NamespaceFixtures.GetNamespaces(context.Background())
+	assert.Nil(s.T(), err)
+
+	testData := []struct {
+		name    string
+		ID      string
+		request *request.Namespace
+	}{
+		{
+			name: "UpdateNamespaceWithNotFoundID",
+			ID:   "10",
+			request: &request.Namespace{
+				Code:        "testUpdated",
+				Description: "test namespace updated",
+			},
+		},
+		{
+			name: "UpdateNamespaceWithEmptyID",
+			ID:   "",
+			request: &request.Namespace{
+				Code:        "testUpdated",
+				Description: "test namespace updated",
+			},
+		},
+		{
+			name: "UpdateNamespaceWithInvalidID",
+			ID:   "InvalidID",
+			request: &request.Namespace{
+				Code:        "testUpdated",
+				Description: "test namespace updated",
+			},
+		},
+		{
+			name: "UpdateNamespaceWithDuplicatedCode",
+			ID:   "2",
+			request: &request.Namespace{
+				Code:        "default",
+				Description: "test namespace updated",
+			},
+		},
+	}
+	for _, tt := range testData {
+		s.T().Run(tt.name, func(t *testing.T) {
+			assert.Nil(
+				s.T(),
+				s.AdminClient.WithMethod(
+					http.MethodPut,
+				).WithRequest(
+					tt.request,
+				).DoRequest(
+					"/namespaces/%s", tt.ID,
+				),
+			)
+		})
+		updatedNamespaces, err := s.NamespaceFixtures.GetNamespaces(context.Background())
```

Comment:
and here we can call -> `actualNamespaces`? probably nothing just happened, so this is not `updatedNamespaces`.